allennlp/training/model.py

```diff
@@ -13,9 +13,9 @@ class Model(torch.nn.Module):
     data generation, because model inputs are named.
     """
 
-    def forward(self,
+    def forward(self,  # pylint: disable=arguments-differ
                 inputs: Dict[str, torch.Tensor],
-                compute_loss: bool = False):  # pylint: disable=arguments-differ
+                compute_loss: bool = False) -> (Union[torch.Tensor, List[torch.Tensor]], Optional[float]):
```

Comment:
I think you can also remove `compute_loss` - we just always compute the loss if the targets are given, and we return `None` if they are not.